src/main/java/techreborn/blockentity/machine/tier0/block/BlockBreakerBlockEntity.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of TechReborn, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2020 TechReborn
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package techreborn.blockentity.machine.tier0.block;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+import reborncore.client.screen.builder.BlockEntityScreenHandlerBuilder;
+import reborncore.client.screen.builder.ScreenHandlerBuilder;
+import reborncore.common.blockentity.MachineBaseBlockEntity;
+import reborncore.common.recipes.RecipeCrafter;
+import reborncore.common.screen.BuiltScreenHandler;
+import reborncore.common.screen.BuiltScreenHandlerProvider;
+import reborncore.common.util.RebornInventory;
+import techreborn.blockentity.machine.GenericMachineBlockEntity;
+import techreborn.blockentity.machine.tier0.block.blockbreaker.BlockBreakerProcessor;
+import techreborn.config.TechRebornConfig;
+import techreborn.init.TRBlockEntities;
+import techreborn.init.TRContent;
+
+/**
+ * <b>A machine that can break the block physically located in front of it</b>
+ * <br>
+ * May not break blocks with hardness < 0, may not break blocks that has no item.
+ * Shares most functionality with other GenericMachineBlockEntities, however does not make use of the {@code RecipeCrafter}
+ *
+ * @author SimonFlapse
+ */
+public class BlockBreakerBlockEntity extends GenericMachineBlockEntity implements BuiltScreenHandlerProvider, BlockProcessable {
+
+	public static final int BASE_BREAK_TIME = TechRebornConfig.blockBreakerBaseBreakTime;
+	public static final int COST_TO_BREAK = TechRebornConfig.blockBreakerEnergyPerTick;
```

Comment:
Be careful doing this, as the config values may be loaded (via reflection) after this value has been set. I would just use the `TechRebornConfig.blockBreakerBaseBreakTime` directly.

I expect this may be an issue in other places, dont worry about those for now.

Reply:
Thanks for pointing it out.

It was just the quickest way to support configurable values but I see the issue.

I'll do the refactoring needed to fix it.